mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -353,3 +354,55 @@ def __init__(self, main_window: MainWindowView) -> None:
         nan_check_menu = [("Crop Coordinates", lambda: main_window.presenter.show_operation("Crop Coordinates")),
                           ("NaN Removal", lambda: main_window.presenter.show_operation("NaN Removal"))]
         self.image.enable_nan_check(actions=nan_check_menu)
+
+
+class MIPlotItem(PlotItem):
+    connect: str | np.ndarray = 'auto'
+    pen_dict: dict = {}
+    name_index: int = 0
+
+    def __init__(self, join_plot=True, *args, **kwds) -> None:
+        super().__init__(*args, **kwds)
+        self.join_plot = join_plot
+        self.vb = self.getViewBox()
+        self.menu = self.vb.menu
+        self.plot_menu = self.menu.addMenu("Plotting Options")
+        self.join_choice_group = QActionGroup(self)
+
+        joined_action = QAction('Joined', self.join_choice_group)
+        joined_action.setCheckable(True)
+        joined_action.setObjectName('Joined')
+        joined_action.triggered.connect(self.set_join_plot)
+        joined_action.setChecked(True)
+        self.plot_menu.addAction(joined_action)
+
+        scatter_action = QAction('Scatter', self.join_choice_group)
+        scatter_action.setCheckable(True)
+        scatter_action.setObjectName('Scatter')
+        scatter_action.triggered.connect(self.set_join_plot)
+        self.plot_menu.addAction(scatter_action)
+
+    def plot(self, *args, **kwargs):
+        if 'name' not in kwargs:
+            kwargs['name'] = 'plot_' + str(self.name_index)
+            self.name_index += 1
+        super().plot(*args, **kwargs)
```

Comment:
You could store the original pen in the `plotDataItem` so save needing to using `pen_dict`.

```
new_plot = super().plot(*args, **kwargs)
new_plot.original_pen = new_plot.opts['pen']
```
then when setting it in `set_join_plot()` you could used
```
item.setPen(item.original_pen)
```